crates/bevy_animation/src/lib.rs

```diff
@@ -691,217 +677,271 @@ where
         + tangent_in_end * step_duration * (lerp.powi(3) - lerp.powi(2))
 }
 
-#[allow(clippy::too_many_arguments)]
-fn apply_animation(
-    weight: f32,
-    animation: &mut PlayingAnimation,
-    paused: bool,
-    root: Entity,
-    time: &Time,
-    animations: &Assets<AnimationClip>,
-    names: &Query<&Name>,
-    transforms: &Query<&mut Transform>,
-    morphs: &Query<&mut MorphWeights>,
-    maybe_parent: Option<&Parent>,
-    parents: &Query<(Has<AnimationPlayer>, Option<&Parent>)>,
-    children: &Query<&Children>,
-) {
-    let Some(animation_clip) = animations.get(&animation.animation_clip) else {
-        return;
-    };
-
-    // We don't return early because seek_to() may have been called on the animation player.
-    animation.update(
-        if paused { 0.0 } else { time.delta_seconds() },
-        animation_clip.duration,
-    );
+/// Adds animation support to an app
+#[derive(Default)]
+pub struct AnimationPlugin;
 
-    if animation.path_cache.len() != animation_clip.paths.len() {
-        let new_len = animation_clip.paths.len();
-        animation.path_cache.iter_mut().for_each(|v| v.clear());
-        animation.path_cache.resize_with(new_len, Vec::new);
-    }
-    if !verify_no_ancestor_player(maybe_parent, parents) {
-        warn!("Animation player on {:?} has a conflicting animation player on an ancestor. Cannot safely animate.", root);
-        return;
+impl Plugin for AnimationPlugin {
+    fn build(&self, app: &mut App) {
+        app.init_asset::<AnimationClip>()
+            .register_asset_reflect::<AnimationClip>()
+            .register_type::<AnimationPlayer>()
+            .register_type::<AnimationTarget>()
+            .add_systems(
+                PostUpdate,
+                (
+                    advance_animations,
+                    animate_targets,
+                    expire_animation_transitions,
+                )
+                    .chain()
+                    .before(TransformSystem::TransformPropagate),
+            );
     }
+}
 
-    let mut any_path_found = false;
-    for (path, bone_id) in &animation_clip.paths {
-        let cached_path = &mut animation.path_cache[*bone_id];
-        let curves = animation_clip.get_curves(*bone_id).unwrap();
-        let Some(target) = entity_from_path(root, path, children, names, cached_path) else {
-            continue;
+impl PlayingAnimation {
+    fn apply(
+        &self,
+        clips: &Assets<AnimationClip>,
+        weight: f32,
+        target_context: &mut AnimationTargetContext,
+    ) {
+        let Some(clip) = clips.get(&self.animation_clip) else {
+            warn!(
+                "Couldn't find the animation clip for {:?} ({:?})",
+                target_context.id, target_context.name,
```

Comment:
Should we name this `entity` instead of `id`? It feels there's a potential for confusion with the `AnimationTargetId`.